Subject: Linux-RT 2.6.24-rt1
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
From linux-rt-users-owner@vger.kernel.org Sat Jul 14 04:08:19 2007
Return-Path: <linux-rt-users-owner@vger.kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.1.7-deb (2006-10-05) on debian
X-Spam-Level: 
X-Spam-Status: No, score=0.2 required=5.0 tests=AWL autolearn=unavailable 
	version=3.1.7-deb
Received: from vger.kernel.org (vger.kernel.org [209.132.176.167]) by
	mail.tglx.de (Postfix) with ESMTP id BB2AB65C292; Sat, 14 Jul 2007 04:08:19
	+0200 (CEST)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand id
	S1758741AbXGNCIT (ORCPT <rfc822;jan.altenberg@linutronix.de> + 1 other);
	Fri, 13 Jul 2007 22:08:19 -0400
Received: (majordomo@vger.kernel.org) by vger.kernel.org id
	S1760025AbXGNCIT (ORCPT <rfc822;linux-rt-users-outgoing>); Fri, 13 Jul 2007
	22:08:19 -0400
Received: from rwcrmhc11.comcast.net ([216.148.227.151]:38099 "EHLO
	rwcrmhc11.comcast.net" rhost-flags-OK-OK-OK-OK) by vger.kernel.org with
	ESMTP id S1758741AbXGNCIR (ORCPT <rfc822;linux-rt-users@vger.kernel.org>);
	Fri, 13 Jul 2007 22:08:17 -0400
Received: from sx.thebigcorporation.com ([69.181.45.228]) by comcast.net
	(rwcrmhc11) with ESMTP id <20070714020816m1100999cee>; Sat, 14 Jul 2007
	02:08:17 +0000
Received: from sx.thebigcorporation.com (localhost.localdomain [127.0.0.1])
	by sx.thebigcorporation.com (8.14.1/8.13.8) with ESMTP id l6E28G5M018639;
	Fri, 13 Jul 2007 19:08:16 -0700
Received: (from sven@localhost) by sx.thebigcorporation.com
	(8.14.1/8.14.1/Submit) id l6E28FCT018638; Fri, 13 Jul 2007 19:08:15 -0700
X-Authentication-Warning: sx.thebigcorporation.com: sven set sender to
	sven@thebigcorporation.com using -f
Subject: Re: [PATCH -rt 6/6] Compile fix for PREEMPT_TIMING on and
	IRQSOFF_TIMING off
From:	Sven-Thorsten Dietrich <sven@thebigcorporation.com>
To:	Kevin Hilman <khilman@mvista.com>
Cc:	tglx@linutronix.de, mingo@elte.hu, linux-rt-users@vger.kernel.org, linux-kernel@vger.kernel.org
In-Reply-To: <46980935.3060509@mvista.com>
References: <20070713175214.336577416@mvista.com>
	 <20070713175229.239602308@mvista.com>  <46980935.3060509@mvista.com>
Content-Type: text/plain
Organization: The Big Corporation
Date:	Fri, 13 Jul 2007 19:08:14 -0700
Message-Id: <1184378894.16207.14.camel@sx.thebigcorporation.com>
Mime-Version: 1.0
X-Mailer: Evolution 2.10.2 (2.10.2-3.fc7) 
Sender:	linux-rt-users-owner@vger.kernel.org
Precedence: bulk
X-Mailing-List:	linux-rt-users@vger.kernel.org
X-Filter-To: .Kernel.rt-users
X-Evolution-Source: imap://tglx%40linutronix.de@localhost:8993/
Content-Transfer-Encoding: 8bit

On Fri, 2007-07-13 at 16:22 -0700, Kevin Hilman wrote:
> [Minor update to avoid a compiler warning in the case of DEBUG_KERNEL=n]
> 

The resent patch (v2) had white space damage - 

Here is a reconstituted version that applies for me on 2.6.22-rt3

Acked-by: Sven-Thorsten Dietrich <sven@thebigcorporation.com>

>From linux-rt-users-owner@vger.kernel.org Fri Jul 13 16:22:34 2007
Return-Path: <linux-rt-users-owner@vger.kernel.org>
Received: from sx.thebigcorporation.com ([unix socket]) by
	sx.thebigcorporation.com (Cyrus v2.3.8-Fedora-RPM-2.3.8-3.fc7) with LMTPA;
	Fri, 13 Jul 2007 16:22:34 -0700
X-Sieve: CMU Sieve 2.3
Received: from vger.kernel.org (vger.kernel.org [209.132.176.167]) by
	sx.thebigcorporation.com (8.14.1/8.13.8) with ESMTP id l6DNMXP6017382 for
	<sven@thebigcorporation.com>; Fri, 13 Jul 2007 16:22:33 -0700
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand id
	S1759195AbXGMXW1 (ORCPT <rfc822;sven@thebigcorporation.com>); Fri, 13 Jul
	2007 19:22:27 -0400
Received: (majordomo@vger.kernel.org) by vger.kernel.org id
	S1760623AbXGMXW1 (ORCPT <rfc822;linux-rt-users-outgoing>); Fri, 13 Jul 2007
	19:22:27 -0400
Received: from h155.mvista.com ([63.81.120.158]:7301 "EHLO
	gateway-1237.mvista.com" rhost-flags-OK-FAIL-OK-OK) by vger.kernel.org with
	ESMTP id S1759161AbXGMXW1 (ORCPT <rfc822;linux-rt-users@vger.kernel.org>);
	Fri, 13 Jul 2007 19:22:27 -0400
Received: from [127.0.0.1] (asshur.mvista.com [10.0.0.11]) by
	hermes.mvista.com (Postfix) with ESMTP id 69F871DE39; Fri, 13 Jul 2007
	16:22:25 -0700 (PDT)
Message-ID: <46980935.3060509@mvista.com>
Date:	Fri, 13 Jul 2007 16:22:29 -0700
From: Kevin Hilman <khilman@mvista.com>
User-Agent: Thunderbird 1.5.0.12 (X11/20070604)
MIME-Version: 1.0
To: Kevin Hilman <khilman@mvista.com>
Cc: tglx@linutronix.de, mingo@elte.hu, linux-rt-users@vger.kernel.org, linux-kernel@vger.kernel.org
Subject: Re: [PATCH -rt 6/6] Compile fix for PREEMPT_TIMING on and
	IRQSOFF_TIMING off
References: <20070713175214.336577416@mvista.com>
	 <20070713175229.239602308@mvista.com>
In-Reply-To: <20070713175229.239602308@mvista.com>
Content-Type: text/plain; charset=ISO-8859-1
Sender: linux-rt-users-owner@vger.kernel.org
Precedence: bulk
X-Mailing-List:	linux-rt-users@vger.kernel.org
X-Evolution-Source: imap://sven@sx.thebigcorporation.com/
Content-Transfer-Encoding: 8bit

[Minor update to avoid a compiler warning in the case of DEBUG_KERNEL=n]

Compile fix for PREEMPT_TIMING on and IRQSOFF_TIMING off

The per-cpu trace_cpu_idle variable is used when timing *either*
IRQs-off or preempt sections.

Signed-off-by: Kevin Hilman <khilman@mvista.com>
---
 kernel/latency_trace.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

Index: linux-2.6.24-rt1/kernel/latency_trace.c
===================================================================
--- linux-2.6.24-rt1.orig/kernel/latency_trace.c	2008-01-25 15:07:19.000000000 -0500
+++ linux-2.6.24-rt1/kernel/latency_trace.c	2008-01-25 15:07:19.000000000 -0500
@@ -85,9 +85,13 @@ static inline int DEBUG_WARN_ON(int cond
 }
 #endif
 
+#if defined(CONFIG_CRITICAL_IRQSOFF_TIMING) || \
+   (defined(CONFIG_CRITICAL_PREEMPT_TIMING) && defined(CONFIG_TRACE_IRQFLAGS))
+   static DEFINE_PER_CPU(int, trace_cpu_idle);
+#endif
+
 #ifdef CONFIG_CRITICAL_IRQSOFF_TIMING
 # ifdef CONFIG_CRITICAL_PREEMPT_TIMING
-   static DEFINE_PER_CPU(int, trace_cpu_idle);
 #  define irqs_off_preempt_count() (preempt_count() && !__get_cpu_var(trace_cpu_idle))
 # else
 #  define irqs_off_preempt_count() 0
